inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -52,6 +53,13 @@ class XmlVisitor : public ngraph::AttributeVisitor {
         return joinVec(adapter.get(), std::string(","));
     }
 
+    std::string translate_attribute_value(std::string name) {
+        // IR format expects attribute values in lower case
+        std::transform(name.begin(), name.end(), name.begin(),
```

Comment:
Do all visit_attribute methods work correct with attributes in lower case?
Or can we get some issues with case sensitive?

Reply:
So the real question is if we have some attributes written in different than lower case in specification (hence it should be in different case in serialized XML). I don't have definite answer to this - I went through different ops spec and it seems that all attributes are lower case (and if not it's a bug in spec), but I could missed something. If we can make this assumption about spec, then we can convert every attribute to lowercase and fix everything at once. Alternative is to convert attribute names one by one when we identify that it is not aligned to spec (e.g. in case of squeezenet model it is auto_broadcast & auto_pad). 

Either way we can have some differences detected during E2E tests. The question is what will produce more bugs. The case with attribute names as presented by VisitorAPI or the case where we convert everything to lowercase.

Ideally, I would propose to make make a rule in spec that attributes names must all be lower case. It would make writing spec easier, and also would prove our code correct. Then if we would found in spec that some attribute is different case (e.g. True/False) we would fix it in spec.

Reply:
@jdanieck I think will be more right to fix behaviour with attributes naming in the AttributeVisitors.
For example here: https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/op/util/attr_types.cpp#L34

Reply:
I considered this approach first but came to the conclusion that I can broke client code which can use old names. 

Example from tests:
jdanieck@ubuntu-ngraph:~/openvino$ ag NUMPY
`ngraph/test/type_prop/broadcast.cpp
446:// NUMPY MODE
454:    auto bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
459:    bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
470:    auto bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
476:    bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
487:    auto bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
492:    bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
501:    const auto bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
511:    auto bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
517:    bc = make_shared<TypeParam>(data, target_shape_const, "NUMPY");
533:    auto bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
540:    bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
547:    bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
555:    bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
569:        auto bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
587:        auto bc = make_shared<TypeParam>(data, target_shape, "NUMPY");
605:        auto bc = make_shared<TypeParam>(data, target_shape, "NUMPY");`

Reply:
We have a talk and cleared concerns about breaking client code - we don't see any risk here. We will go with approach Ilya suggested and fix attribute names in nGraph code.

Reply:
Just btw - this `translate_type_name` contains a lot of unnecessary copying and its return value is used as rvalue at the call site. I suggest to change it the following way:

```
void translate_type_name(std::string& name) {
    // assign a translated value to name or do nothing
}

std::string node_type_name{node->get_type_name()};
translate_type_name(node_type_name);
layer_type_attribute.set_value(node_type_name.c_str());
```

Reply:
RVO won't work in this case?

Reply:
Because the result of this function is used as rvalue - I'm not sure. There's definitely an unnecessary copy of the argument. If you change it to an output param version (which is not so great but might affect the performance) there's no copying at all.

Reply:
ACK. I've removed this funcion, so currently there is no work. Thanks for pointing this out.